src/SigasiProjectCreator/Creator.py

```diff
@@ -395,14 +402,19 @@ def add_mapping(self, path, library):
     def unmap(self, path):
         self.__libraryMappingFileCreator.unmap(posixpath(path))
 
+    def add_verilog_include(self, path):
+        self.verilog_includes.append(path)
+
     def write(self, destination, force_vhdl=None, force_verilog=None, verilog_includes=None, verilog_defines=None, force_vunit=None):
         self.__projectFileCreator.write(destination, force_vhdl, force_verilog, force_vunit)
         self.__libraryMappingFileCreator.write(destination)
         for projectVersionCreator in self.__projectVersionCreators:
             projectVersionCreator.write(destination)
-        if ((verilog_includes is not None) and (len(verilog_includes) > 0)) or ((verilog_defines is not None) and (len(verilog_defines) > 0)):
-                verilog_prefs = ProjectPreferencesCreator('verilog', verilog_includes, verilog_defines)
-                verilog_prefs.write(destination)
+        self.verilog_includes.extend(verilog_includes)
+        if ((self.verilog_includes is not None) and (len(self.verilog_includes) > 0)) or \
+                ((verilog_defines is not None) and (len(verilog_defines) > 0)):
```

Comment:
if you're sure `verilog_defines` is a list, this is equivalent to simply `verilog_defines` (`if [] ...` and `if None ...` both evaluate to false).

Reply:
Updated as suggested